connector/src/main/java/org/geysermc/connector/configuration/GeyserJacksonConfiguration.java

```diff
@@ -183,26 +184,30 @@ public int getCompressionLevel() {
         }
     }
 
-    @Getter
     @JsonIgnoreProperties(ignoreUnknown = true)
     public static class RemoteConfiguration implements IRemoteConfiguration {
-        @Setter
+        @Getter @Setter
         @AsteriskSerializer.Asterisk(isIp = true)
         private String address = "auto";
 
-        @Setter
+        @Getter @Setter
         private int port = 25565;
 
-        @Setter
         @JsonProperty("auth-type")
-        private String authType = "online";
+        private String authTypeName = "online";
+
+        @Getter @Setter
+        private AuthType authType = AuthType.getByName(authTypeName);
```

Comment:
I would implement this with the same method used in https://github.com/GeyserMC/Geyser/blob/master/connector/src/main/java/org/geysermc/connector/configuration/EmoteOffhandWorkaroundOption.java. That way we don't need to add another variable and instead just deserialize directly to the AuthType.